StressLogAnalyzer is failing to build

```
  [ 98%] Built target runtimeinfo
  [100%] Built target singlefilehost
  Killed
  make[2]: *** [tools/StressLogAnalyzer/CMakeFiles/StressLogAnalyzer.dir/build.make:89: tools/StressLogAnalyzer/CMakeFiles/StressLogAnalyzer.dir/StressLogPlugin.cpp.o] Error 137
  make[1]: *** [CMakeFiles/Makefile2:5167: tools/StressLogAnalyzer/CMakeFiles/StressLogAnalyzer.dir/all] Error 2
  make: *** [Makefile:130: all] Error 2
  ~/github/wfurt-runtime/src/coreclr
  Failed to build "CoreCLR component".
/home/furt/github/wfurt-runtime/src/coreclr/runtime.proj(59,5): error MSB3073: The command ""/home/furt/github/wfurt-runtime/src/coreclr/build-runtime.sh" -x64 -release -os Linux -pgodatapath "/home/furt/.nuget/packages/optimization.linux-x64.pgo.coreclr/1.0.0-prerelease.21577.2"" exited with code 2.

```
I just sync main and my build suddenly start failing. Turns out that 4G VM is no longer good enough to build. So I added 2G more and it still fails when clang reaches 5.5G. I can probably keeping more ram but the spike in needed is very large. If I need to do that on every VM I have that would generally kill my dev machine. 

I'm wondering if there is anything we can do
cc: @am11 @jkotas @janvorli and @cshung who recently touched this ara.

> I wonder if there is anything we can do

StressLogAnalyzer is meant to be a debugging tool only. If it is necessary, we can take it off the build.
When I merged though, the build was fine for the whole CI matrix, so not sure what is wrong there.

> 4G VM is not enough

It shouldn't take that much memory to compile that executable. Excluding its dependencies (i.e. the PAL), the executable has just a few source files with an overall less than 10K lines.

The log looks somewhat familiar:

```
  Killed
  ...
  make[1]: *** [CMakeFiles/Makefile2:5167: tools/StressLogAnalyzer/CMakeFiles/StressLogAnalyzer.dir/all] Error 2
  make: *** [Makefile:130: all] Error 2
```

I observe the same thing with Alpine Linux compiling the StressLogAnalyzer. I thought it was Alpine specific and therefore I intentionally excluded Alpine from the build and think it is some Alpine specific stuff. Given the observation from @am11, it could be the clang version that is causing the issue.

--- a/src/coreclr/tools/StressLogAnalyzer/StressLogPlugin.cpp
+++ b/src/coreclr/tools/StressLogAnalyzer/StressLogPlugin.cpp
@@ -373,7 +373,7 @@ bool FilterMessage(StressLog::StressLogHeader* hdr, ThreadStressLog* tsl, uint32
     {
     case    IS_UNINTERESTING:
     case    IS_UNKNOWN:
-        break;    
+        break;
     case    IS_THREAD_WAIT:
     case    IS_THREAD_WAIT_DONE:
         RememberThreadForHeap(tsl->threadId, (int64_t)args[0], GC_THREAD_FG);
@@ -869,7 +869,7 @@ bool ParseOptions(int argc, char* argv[])
                         printf("-f:<format string> expected\n");
                         return false;
                     }
-                    
+
                     // remove double quotes around the string, if given
                     if (actualSize >= 2 && buf[0] == '"' && buf[actualSize - 1] == '"')
                     {
@@ -1234,7 +1234,13 @@ int ProcessStressLog(void* baseAddress, int argc, char* argv[])
     StressLog::writing_base_address = (size_t)hdr->memoryBase;
     StressLog::reading_base_address = (size_t)baseAddress;
     s_hdr = hdr;
-    s_threadMsgBuf = new StressThreadAndMsg[MAX_MESSAGE_COUNT];
+
+    // Workaround for clang SIGKILL (exit code 137) crash.  Apparently, clang does not
+    // like large array instantiation on R.H.S of static variable. It was reproduced on
+    // linux-x64 clang v6, v10 and v13.
+    auto temp = new StressThreadAndMsg[MAX_MESSAGE_COUNT];
+    s_threadMsgBuf = temp;
+
     int threadStressLogIndex = 0;
     double latestTime = FindLatestTime(hdr);
     if (s_timeFilterStart < 0)
@@ -1278,7 +1284,7 @@ int ProcessStressLog(void* baseAddress, int argc, char* argv[])
     // the interlocked increment may have increased s_msgCount beyond MAX_MESSAGE_COUNT -
     // make sure we don't go beyond the end of the buffer
     s_msgCount = min(s_msgCount, MAX_MESSAGE_COUNT);
-    
+
     if (s_gcFilterStart != 0)
     {
         // find the time interval that includes the GCs in question
@@ -1363,7 +1369,7 @@ int ProcessStressLog(void* baseAddress, int argc, char* argv[])
         s_msgCount = remMsgCount;
     }
 
-    // if the sort becomes a bottleneck, we can do a bucket sort by time 
+    // if the sort becomes a bottleneck, we can do a bucket sort by time
     // (say fractions of a second), then sort the individual buckets,
     // perhaps on multiple threads
     qsort(s_threadMsgBuf, s_msgCount, sizeof(s_threadMsgBuf[0]), CmpMsg);
@@ -1454,7 +1460,7 @@ int ProcessStressLog(void* baseAddress, int argc, char* argv[])
 
     ptrdiff_t usedSize = hdr->memoryCur - hdr->memoryBase;
     ptrdiff_t availSize = hdr->memoryLimit - hdr->memoryCur;
-    printf("Used file size: %6.3f GB, still available: %6.3f GB, %d threads total, %d overwrote earlier messages\n", 
+    printf("Used file size: %6.3f GB, still available: %6.3f GB, %d threads total, %d overwrote earlier messages\n",
         (double)usedSize / (1024 * 1024 * 1024), (double)availSize/ (1024 * 1024 * 1024),
         s_threadStressLogCount, (int)s_wrappedWriteThreadCount);
     if (hdr->threadsWithNoLog != 0)

